Lesson6/Task2/Program.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Reflection;
+using System.IO;
+
+
+namespace Task2
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Assembly assembly = null;
+
+            try
+            {
+                assembly = Assembly.LoadFrom("Task1.dll");
+            }
+            catch (FileNotFoundException e)
+            {
+                Console.WriteLine(e.Message);
+            }
+            Type type = assembly.GetType("Task1.CelsiusToFahrenheit");
+
+            object instance = Activator.CreateInstance(type);
```

Comment:
Try doing this in a way with shared library with interface:
1. create shared library e.g. `Converters.Interfaces` 
2. define an interface `ITemperatureConverter`
3. add reference to this library from Task2
4. add reference to this library from Task1 and implement an interface inside Task1 library
5. here in Task2 you can dynamically load Task1 library (as you do right now), get implementers of `ITemperatureConverter` interface, create an instace of one of them and cast it to `ITemperatureConverter` type
6. Then you can use this instance as a regular object and call methods directly without reflection